packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx

```diff
@@ -205,11 +216,38 @@ const MetricsVisualization: FC<Props> = ({
         };
     }, [activeData]);
 
+    const segmentedData = useMemo((): {
+        segment: string | null;
+        color: string;
+        data: MetricsExplorerQueryResults['results'];
+    }[] => {
+        const segmentsMap = new Map<
+            string | null,
+            MetricsExplorerQueryResults['results']
+        >();
+
+        activeData.forEach((row) => {
+            const segment = row.segment; // Preserve `null` as is
+            if (!segmentsMap.has(segment)) {
+                segmentsMap.set(segment, []);
+            }
+            segmentsMap.get(segment)!.push(row);
+        });
+
+        return Array.from(segmentsMap.entries()).map(
+            ([segment, segmentData], i) => ({
+                segment,
+                color: colors[CHART_COLORS[i % CHART_COLORS.length]][6],
```

Comment:
```suggestion
                color: colors[CHART_COLORS[i % CHART_COLORS.length]]?.[6],
```

try unique payment count and then group by order_id